components/SortFilter.tsx

```diff
@@ -0,0 +1,401 @@
+import { SyntheticEvent, useMemo, useState } from 'react';
+import { Menu } from '@headlessui/react';
+import { useDebounce, useLocation } from 'react-use';
+import { Disclosure } from '@headlessui/react';
+import { IconCaret, IconFilters, IconXMark } from './Icon';
+import { Heading, Text } from './Text';
+import { Link } from './Link';
+import { Filter, Collection, FilterType } from '@/lib/shopify/types';
+import { useRouter } from 'next/router';
+import { LocationSensorState } from 'react-use/lib/useLocation';
+
+export type AppliedFilter = {
+  label: string;
+  urlParam: {
+    key: string;
+    value: string;
+  };
+};
+
+export type SortParam =
+  | 'price-low-high'
+  | 'price-high-low'
+  | 'best-selling'
+  | 'newest'
+  | 'featured';
+
+type Props = {
+  filters: Filter[];
+  appliedFilters?: AppliedFilter[];
+  children: React.ReactNode;
+  collections?: Collection[];
+};
+
+export function SortFilter({
+  filters,
+  appliedFilters = [],
+  children,
+  collections = [],
+}: Props) {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <>
+      <div className="flex items-center justify-between w-full">
+        <button
+          onClick={() => setIsOpen(!isOpen)}
+          className={
+            'relative flex items-center justify-center w-8 h-8 focus:ring-primary/5'
+          }
+        >
+          <IconFilters />
+        </button>
+        <SortMenu />
+      </div>
+      <div className="flex flex-col flex-wrap md:flex-row">
+        <div
+          className={`transition-all duration-200 ${isOpen
```

Comment:
```suggestion
          className={clsx('transition-all duration-200 opacity-0 md:min-w-[0px] md:w-[0px] pr-0 max-h-0 md:max-h-full', {'opacity-100 min-w-full md:min-w-[240px] md:w-[240px] md:pr-8 max-h-full': isOpen})}
```

Please, can we use clsx. This is also based on the contribution guide as well I think.